python/mp/rosary.py

```diff
@@ -29,25 +30,54 @@ def copy_color(self, color):
         self.color = copy.copy(color)
 
 
+class Updater:
+    def __init__(self, name='', bead_list=[], osc_client=None):
+        self.name = name
+        self.bead_list = bead_list
+        self.osc_client=osc_client
+        
+    def update(self):
+        msg = osc_message_builder.OscMessageBuilder(address = "/bead")
+        msg.add_arg(self.name)                 # name (class)
+        msg.add_arg(int(0))                    # base
+        msg.add_arg(int(len(self.bead_list)))  # length
+        
+        payload = bytearray()
+        for bead in self.bead_list:
+            payload.extend(struct.pack('!HHHH',
+                                       int(bead.color.r * 0xffff),
+                                       int(bead.color.g * 0xffff),
+                                       int(bead.color.b * 0xffff),
+                                       int(bead.color.brightness)
+            ))
```

Comment:
We're sending the starting bead, the number of beads and then a binary blob that contains the colors for all of those beads. It means we send 3 updates for all the "beads": 1 for the original rosary beads, 1 for the lights in the 9 base structures, and 1 for the 480 LEDs in the cross. It was too expensive to send 60 + 9 + 480 individual updates.